File: app/cpp/unix/code/src/manager/GProcess.cpp
//===============================================
#include "GProcess.h"
#include "GEnv.h"
#include "GSocket.h"
#include "GTest.h"
//===============================================
GProcess::GProcess()
: GObject() {

}
//===============================================
GProcess::~GProcess() {

}
//===============================================
void GProcess::run(int _argc, char** _argv) {
    bool lIsTestEnv = GENV->isTestEnv();
    if(lIsTestEnv) {
        runTest(_argc, _argv);
    }
    else {
        runProd(_argc, _argv);
    }
}
//===============================================
void GProcess::runTest(int _argc, char** _argv) {
    GTest lTest;
    lTest.run(_argc, _argv);
    m_logs.addLogs(lTest.m_logs);
}
//===============================================
void GProcess::runProd(int _argc, char** _argv) {

}
//===============================================

File: app/python/c/unix/code/src/py/manager/GProcess.py
#================================================
from functions import *
from manager.GObject import GObject
from manager.GTest import GTest
#================================================
class GProcess(GObject):
    #================================================
    def __init__(self):
        GObject.__init__(self)
    #================================================
    def init(self):
        pass
    #================================================
    def clean(self):
        pass
    #================================================
    def run(self):
        lModule = ""
        if len(sys.argv) > 1:
            lModule = sys.argv[1]
            
        if lModule == "":
            self.m_logs.addError("Le module est obligatoire.");
        elif lModule == "test":
            self.runTest();
        else:
            self.m_logs.addError("Le module est inconnu.");
    #================================================
    def runTest(self):
        lObj = GTest()
        lObj.run()
        self.m_logs.addLogs(lObj.m_logs)
#================================================